### PR TITLE
nested-stacks

### PR DESCRIPTION
**Pull request by snjkumar23**
## What does this PR do and why?

Instructions for cloud9 integration

## PR acceptance checklist

This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [ ] Added/Updated documentation if applicable.
- [x] Pre-commit checks passed.
- [x] Lint and Nag checks passed.
- [ ] If releasing a new version, have you bumped the version `make version part=<major|minor|patch>`?

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: content/intermediate/templates/nested-stacks/index.md [hunk 1]
```diff
@@ -221,23 +258,56 @@ IamStack:
...
-
+:::::tabs{variant="container"}
+::::tab{id="cloud9" label="Cloud9"}
+1. In the **Cloud9 terminal** navigate to `code/workspace/nested-stacks`:
+  :::code{language=shell showLineNumbers=false showCopyAction=true}
+  cd cfn101-workshop/code/workspace/nested-stacks
+  :::
+1. Use the AWS CLI to update the stack. The required parameter `--template-body` have been pre-filled for you. Replace the `ParameterValue` **bucketName** with the value you have written down in [Prepare S3 bucket](#2-prepare-s3-bucket) section. Replace the `ParameterValue` **AZ1** and  **AZ2** with the value of any 2 AZ's from your vpc.
+  :::code{language=shell showLineNumbers=false showCopyAction=true}
+  aws cloudformation update-stack --stack-name cfn-workshop-nested-stacks --template-body file://main.yaml --parameters ParameterKey="S3BucketName",ParameterValue="bucketName" ParameterKey="AvailabilityZones",ParameterValue=AZ1\\,AZ2 --capabilities CAPABILITY_NAMED_IAM
```
Comment (cjhar) [inline content/intermediate/templates/nested-stacks/index.md:291]:
```suggestion
  aws cloudformation update-stack --stack-name cfn-workshop-nested-stacks \
  --template-body file://main.yaml \
  --parameters ParameterKey="S3BucketName",ParameterValue="bucketName" ParameterKey="AvailabilityZones",ParameterValue=AZ1\\,AZ2 \
  --capabilities CAPABILITY_NAMED_IAM
```

File: content/intermediate/templates/nested-stacks/index.md [hunk 2]
```diff
@@ -426,23 +496,61 @@ Before you can deploy the updated nested stack, you must update the templates in
...
 1. In **Template source**, choose **Upload a template file**.
 1. Click on **Choose file** button and navigate to your workshop directory.
 1. Select the `main.yaml` template file and click **Next**.
-1. Follow the wizard, acknowledge IAM capabilities and click on **Update stack**.
+1. Follow the wizard, acknowledge IAM capabilities and click on **Submit**.
+
+::::
+:::::
 
 #### 7. Test the deployment
```
Comment (cjhar) [inline content/intermediate/templates/nested-stacks/index.md:555]: Note in the Session Manager lab we don't describe how to do this, we use an outdated animated gif. I suggest we just link to the console on line 571
```
Verify that you can log in to the instance via **[Session Manager](https://us-east-1.console.aws.amazon.com/systems-manager/session-manager/sessions?region=us-east-1)** and select your region.
```

File: content/intermediate/templates/nested-stacks/index.md [hunk 3]
```diff
@@ -426,23 +496,61 @@ Before you can deploy the updated nested stack, you must update the templates in
...
 
+:::::tabs{variant="container"}
+::::tab{id="cloud9" label="Cloud9"}
+1. In the **Cloud9 terminal** navigate to `code/workspace/nested-stacks`:
+  :::code{language=shell showLineNumbers=false showCopyAction=true}
+  cd cfn101-workshop/code/workspace/nested-stacks
+  :::
+1. Use the AWS CLI to update the stack. The required parameter `--template-body` have been pre-filled for you. Replace the `ParameterValue` **bucketName** with the value you have written down in [Prepare S3 bucket](#2-prepare-s3-bucket) section. Replace the `ParameterValue` **AZ1** and  **AZ2** with the value of any 2 AZ's from your vpc.
+  :::code{language=shell showLineNumbers=false showCopyAction=true}
+  aws cloudformation update-stack --stack-name cfn-workshop-nested-stacks --template-body file://main.yaml --parameters ParameterKey="S3BucketName",ParameterValue="bucketName" ParameterKey="AvailabilityZones",ParameterValue=AZ1\\,AZ2 --capabilities CAPABILITY_NAMED_IAM
```
Comment (cjhar) [inline content/intermediate/templates/nested-stacks/index.md:533]:
```suggestion
  aws cloudformation update-stack --stack-name cfn-workshop-nested-stacks \
  --template-body file://main.yaml \
  --parameters ParameterKey="S3BucketName",ParameterValue="bucketName" \
  ParameterKey="AvailabilityZones",ParameterValue=AZ1\\,AZ2 \
  --capabilities CAPABILITY_NAMED_IAM
```

File: content/intermediate/templates/nested-stacks/index.md [hunk 4]
```diff
@@ -426,23 +496,61 @@ Before you can deploy the updated nested stack, you must update the templates in
...
+  :::
+1. If the `update-stack` command was successfully sent, CloudFormation will return `StackId`.
+  :::code{language=shell showLineNumbers=false showCopyAction=false}
```
Comment (cjhar) [inline content/intermediate/templates/nested-stacks/index.md:536]:
```suggestion
  :::code{language=json showLineNumbers=false showCopyAction=false}
```

File: content/intermediate/templates/nested-stacks/index.md [hunk 5]
```diff
@@ -468,7 +576,7 @@ If you not sure how to do that, follow the instructions from the [Session Manage
 
 Follow these steps to clean up created resources:
 
-1. In the **[CloudFormation console](https://console.aws.amazon.com/cloudformation)**, select the **root** stack you have created in this lab. For example `cfn-workshop-nested-stack`.
+1. In the **[CloudFormation console](https://console.aws.amazon.com/cloudformation)**, select the **root** stack you have created in this lab. For example `cfn-workshop-nested-stacks`.
 1. The **root** stack will handle the deletion of all the **children** stacks for you.
 1. In the top right corner, click on **Delete**.
 1. In the pop-up window click on **Delete stack**.
```
Comment (cjhar) [inline content/intermediate/templates/nested-stacks/index.md:582]:
```suggestion
1. In the pop-up window click on **Delete**.
```

Should we also suggest emptying/deleting the S3 bucket used for templates?

File: content/intermediate/templates/nested-stacks/index.md [hunk 6]
```diff
@@ -426,23 +496,61 @@ Before you can deploy the updated nested stack, you must update the templates in
...
 1. In **Template source**, choose **Upload a template file**.
 1. Click on **Choose file** button and navigate to your workshop directory.
 1. Select the `main.yaml` template file and click **Next**.
-1. Follow the wizard, acknowledge IAM capabilities and click on **Update stack**.
+1. Follow the wizard, acknowledge IAM capabilities and click on **Submit**.
+
+::::
+:::::
 
 #### 7. Test the deployment
```
Comment (cjhar) [inline content/intermediate/templates/nested-stacks/index.md:556]: suggest updating image at line 455 to use us-east-1, I will update this

File: content/intermediate/templates/nested-stacks/index.md [hunk 7]
```diff
@@ -426,23 +496,61 @@ Before you can deploy the updated nested stack, you must update the templates in
...
+  :::
+1. Use the AWS CLI to update the stack. The required parameter `--template-body` have been pre-filled for you. Replace the `ParameterValue` **bucketName** with the value you have written down in [Prepare S3 bucket](#2-prepare-s3-bucket) section. Replace the `ParameterValue` **AZ1** and  **AZ2** with the value of any 2 AZ's from your vpc.
+  :::code{language=shell showLineNumbers=false showCopyAction=true}
+  aws cloudformation update-stack --stack-name cfn-workshop-nested-stacks --template-body file://main.yaml --parameters ParameterKey="S3BucketName",ParameterValue="bucketName" ParameterKey="AvailabilityZones",ParameterValue=AZ1\\,AZ2 --capabilities CAPABILITY_NAMED_IAM
+  :::
+1. If the `update-stack` command was successfully sent, CloudFormation will return `StackId`.
+  :::code{language=shell showLineNumbers=false showCopyAction=false}
+  "StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/cfn-workshop-nested-stacks/739fafa0-e4d7-11ed-a000-12d9009553ff"
+  :::
+1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new tab and check if the stack status is **CREAUPDATETE_COMPLETE**.
```
Comment (cjhar) [inline content/intermediate/templates/nested-stacks/index.md:539]:
```suggestion
1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new tab and check if the stack status is **UPDATE_COMPLETE**.
```

File: content/intermediate/templates/nested-stacks/index.md [hunk 8]
```diff
@@ -221,23 +258,56 @@ IamStack:
...
+  :::
+1. Use the AWS CLI to update the stack. The required parameter `--template-body` have been pre-filled for you. Replace the `ParameterValue` **bucketName** with the value you have written down in [Prepare S3 bucket](#2-prepare-s3-bucket) section. Replace the `ParameterValue` **AZ1** and  **AZ2** with the value of any 2 AZ's from your vpc.
+  :::code{language=shell showLineNumbers=false showCopyAction=true}
+  aws cloudformation update-stack --stack-name cfn-workshop-nested-stacks --template-body file://main.yaml --parameters ParameterKey="S3BucketName",ParameterValue="bucketName" ParameterKey="AvailabilityZones",ParameterValue=AZ1\\,AZ2 --capabilities CAPABILITY_NAMED_IAM
+  :::
+1. If the `update-stack` command was successfully sent, CloudFormation will return `StackId`.
+  :::code{language=shell showLineNumbers=false showCopyAction=false}
+  "StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/cfn-workshop-nested-stacks/739fafa0-e4d7-11ed-a000-12d9009553ff"
+  :::
+1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new tab and check if the stack status is **CREATE_COMPLETE**.
```
Comment (cjhar) [inline content/intermediate/templates/nested-stacks/index.md:297]:
```suggestion
1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new tab and check if the stack status is **UPDATE_COMPLETE**.
```

**Review by cjhar — CHANGES_REQUESTED** [review]
Thanks Sanjeev, please see comments

File: content/intermediate/templates/nested-stacks/index.md (hunk 6 above)
Comment (cjhar) [reply content/intermediate/templates/nested-stacks/index.md:556]: updated image committed

File: content/intermediate/templates/nested-stacks/index.md (hunk 5 above)
Comment (snjkumar23) [reply content/intermediate/templates/nested-stacks/index.md:582]: That bucket was created in some other lab, not sure deleting that now will impact something later. so safer side for now we can skip that.

File: content/intermediate/templates/nested-stacks/index.md [hunk 9]
```diff
@@ -157,19 +173,40 @@ such as `Template format error: [/Resources/VpcStack] resource definition is mal
...
+:::::tabs{variant="container"}
+
+::::tab{id="cloud9" label="Cloud9"}
+1. In the **Cloud9 terminal** navigate to `code/workspace/nested-stacks`:
+  :::code{language=shell showLineNumbers=false showCopyAction=true}
+  cd cfn101-workshop/code/workspace/nested-stacks
+  :::
+1. Use the AWS CLI to create the stack. The required parameter `--template-body` have been pre-filled for you. Replace the `ParameterValue` **bucketName** with the value you have written down in [Prepare S3 bucket](#2-prepare-s3-bucket) section. Replace the `ParameterValue` **AZ1** and  **AZ2** with the value of any 2 AZ's from your vpc.
+  :::code{language=shell showLineNumbers=false showCopyAction=true}
+  aws cloudformation create-stack --stack-name cfn-workshop-nested-stacks --template-body file://main.yaml --parameters ParameterKey="S3BucketName",ParameterValue="bucketName" ParameterKey="AvailabilityZones",ParameterValue=AZ1\\,AZ2 --capabilities CAPABILITY_NAMED_IAM
```
[the remark this answers is not in the document]
Comment (snjkumar23) [reply content/intermediate/templates/nested-stacks/index.md:185]: I have done some changes for this, please review and feel free to modify language if you see some improvement scope

File: content/intermediate/templates/nested-stacks/index.md [hunk 10]
```diff
@@ -144,32 +144,77 @@ VpcStack:
...
+::::tab{id="cloud9" label="Cloud9"}
+1. In the **Cloud9 terminal** navigate to `code/workspace/nested-stacks`:
+  :::code{language=shell showLineNumbers=false showCopyAction=true}
+  cd cfn101-workshop/code/workspace/nested-stacks
+  :::
+1. Use AWS EC2 Api to get list of availability zones from your vpc and copy any 2 AZ's from the output to use them in next step:
+  :::code{language=shell showLineNumbers=false showCopyAction=true}
+  aws ec2 describe-availability-zones --output json --query "AvailabilityZones[*].ZoneName"
+  :::
+1. Use the AWS CLI to create the stack. The required parameter `--template-body` have been pre-filled for you. Replace the `ParameterValue` **bucketName** with the value you have written down in [Prepare S3 bucket](#2-prepare-s3-bucket) section. Replace the `ParameterValue` **AZ1** and **AZ2** which you copied in previous step.
```
Comment (sdangol) [inline content/intermediate/templates/nested-stacks/index.md:188]:
```suggestion
1. Use the AWS CLI to create the stack. The required parameter `--template-body` have been pre-filled for you. Replace the `ParameterValue` **bucketName** with the value you have written down in [Prepare S3 bucket](#2.-prepare-s3-bucket) section. Replace the `ParameterValue` **AZ1** and **AZ2** which you copied in previous step.
```

File: content/intermediate/templates/nested-stacks/index.md [hunk 11]
```diff
@@ -144,32 +144,77 @@ VpcStack:
...
+
+::::tab{id="local" label="Local development"}
 1. Navigate to CloudFormation in the console and click **Create stack With new resources (standard)**.
 1. In **Prepare template** select **Template is ready**.
 1. In **Template source** select **Upload a template file**.
 1. Choose a file `main.yaml`.
-1. Enter a **Stack name**. For example, cfn-workshop-nested-stack
+1. Enter a **Stack name**. For example, cfn-workshop-nested-stacks.
 1. For the **AvailabilityZones** parameter, select 2 AZs.
 1. For the **S3BucketName** provide the name of the bucket you have written down in [Prepare S3 bucket](#2-prepare-s3-bucket) section.
```
Comment (sdangol) [inline content/intermediate/templates/nested-stacks/index.md:209]:
```suggestion
1. For the **S3BucketName** provide the name of the bucket you have written down in [Prepare S3 bucket](#2.-prepare-s3-bucket) section.
```

File: content/intermediate/templates/nested-stacks/index.md [hunk 12]
```diff
@@ -144,32 +144,77 @@ VpcStack:
...
+::::
+:::::
+
+
 
 ##### 4. Deploy VPC Nested Stack
 
 :::alert{type="info"}
 Please note **YAML** is indention sensitive mark down language. If `cfn-lint` or CloudFormation console reports errors,
 such as `Template format error: [/Resources/VpcStack] resource definition is malformed` \
```
Comment (cjhar) [inline content/intermediate/templates/nested-stacks/index.md:172]:
```suggestion
such as `Template format error: [/Resources/VpcStack] resource definition is malformed`, 
```

File: content/intermediate/templates/nested-stacks/index.md [hunk 13]
```diff
@@ -144,32 +144,77 @@ VpcStack:
...
+1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new tab and check if the stack status is **CREATE_COMPLETE**.
+::::
+
+::::tab{id="local" label="Local development"}
 1. Navigate to CloudFormation in the console and click **Create stack With new resources (standard)**.
 1. In **Prepare template** select **Template is ready**.
 1. In **Template source** select **Upload a template file**.
 1. Choose a file `main.yaml`.
-1. Enter a **Stack name**. For example, cfn-workshop-nested-stack
+1. Enter a **Stack name**. For example, cfn-workshop-nested-stacks.
```
Comment (cjhar) [inline content/intermediate/templates/nested-stacks/index.md:207]:
```suggestion
1. Enter a **Stack name**. For example, `cfn-workshop-nested-stacks`.
```

**Review by cjhar — COMMENTED** [review]
Some small changes plus sdangol's changes please. I will revert the wrong image at the end of the lesson

File: content/intermediate/templates/nested-stacks/index.md [hunk 14]
```diff
@@ -426,23 +511,68 @@ Before you can deploy the updated nested stack, you must update the templates in
 Similar to the [uploading the VPC stack](#3-upload-the-vpc-stack-to-s3) in a previous step, upload the `vpc.yaml`, `ec2.yaml`
```
Comment (sdangol) [inline content/intermediate/templates/nested-stacks/index.md:511]:
```suggestion
Similar to the [uploading the VPC stack](#3.-upload-the-vpc-stack-to-s3) in a previous step, upload the `vpc.yaml`, `ec2.yaml`
```

File: content/intermediate/templates/nested-stacks/index.md [hunk 15]
```diff
@@ -426,23 +511,68 @@ Before you can deploy the updated nested stack, you must update the templates in
...
+  aws cloudformation update-stack --stack-name cfn-workshop-nested-stacks \
+  --template-body file://main.yaml \
+  --parameters ParameterKey="S3BucketName",ParameterValue="bucketName" ParameterKey="AvailabilityZones",ParameterValue=AZ1\\,AZ2 \
+  --capabilities CAPABILITY_NAMED_IAM
+  :::
+1. If the `update-stack` command was successfully sent, CloudFormation will return `StackId`.
+  :::code{language=json showLineNumbers=false showCopyAction=false}
+  "StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/cfn-workshop-nested-stacks/739fafa0-e4d7-11ed-a000-12d9009553ff"
+  :::
+1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new tab and check if the stack status is **CREATE_COMPLETE**.
```
Comment (sdangol) [inline content/intermediate/templates/nested-stacks/index.md:561]:
```suggestion
1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new tab and check if the stack status is **UPDATE_COMPLETE**.
```

File: content/intermediate/templates/nested-stacks/index.md
Comment (sdangol) [inline content/intermediate/templates/nested-stacks/index.md:1]: The resource block is missing `SecurityGroupIngress` in line [404-421](https://github.com/aws-samples/cfn101-workshop/pull/225/files#diff-b964c545c069b7d213ba4eee31c6a610205f8a7fe067d3a6b359ab598dece918R404-R421)

**Review by sdangol — CHANGES_REQUESTED** [review]
Added some feedbacks

**Comment by snjkumar23** [comment]
> 

done all

File: content/intermediate/templates/nested-stacks/index.md [hunk 16]
```diff
@@ -423,26 +509,71 @@ Outputs:
...
+::::tab{id="cloud9" label="Cloud9"}
+1. In the **Cloud9 terminal** navigate to `code/workspace/nested-stacks`:
+  :::code{language=shell showLineNumbers=false showCopyAction=true}
+  cd cfn101-workshop/code/workspace/nested-stacks
+  :::
+1. Use AWS EC2 Api to get list of availability zones from your vpc and copy any 2 AZ's from the output to use them in next step:
+  :::code{language=shell showLineNumbers=false showCopyAction=true}
+  aws ec2 describe-availability-zones --output json --query "AvailabilityZones[*].ZoneName"
+  :::
+1. Use the AWS CLI to update the stack. The required parameter `--template-body` have been pre-filled for you. Replace the `ParameterValue` **bucketName** with the value you have written down in [Prepare S3 bucket](#2-prepare-s3-bucket) section. Replace the `ParameterValue` **AZ1** and **AZ2** which you copied in previous step.
```
Comment (sdangol) [inline content/intermediate/templates/nested-stacks/index.md:551]:
```suggestion
1. Use the AWS CLI to update the stack. The required parameter `--template-body` have been pre-filled for you. Replace the `ParameterValue` **bucketName** with the value you have written down in [Prepare S3 bucket](#2.-prepare-s3-bucket) section. Replace the `ParameterValue` **AZ1** and **AZ2** which you copied in previous step.
```

File: content/intermediate/templates/nested-stacks/index.md [hunk 17]
```diff
@@ -144,32 +144,77 @@ VpcStack:
...
+  :::code{language=shell showLineNumbers=false showCopyAction=true}
+  cd cfn101-workshop/code/workspace/nested-stacks
+  :::
+1. Run the below command to copy the `vpc.yaml` to S3 bucket. Replace the `bucket-name` with the bucket name from previous step.
+  :::code{language=shell showLineNumbers=false showCopyAction=true}
+  aws s3 cp vpc.yaml s3://{bucket-name}
+  :::
+::::
+::::tab{id="local" label="Local development"}
 1. Navigate to the [S3 console](https://console.aws.amazon.com/s3/home?region=eu-west-1) and select your bucket.
```
Comment (cjhar) [inline content/intermediate/templates/nested-stacks/index.md:159]:
```suggestion
1. Navigate to the [S3 console](https://console.aws.amazon.com/s3/home) and select your bucket.
```

File: content/intermediate/templates/nested-stacks/index.md [hunk 18]
```diff
@@ -221,23 +266,63 @@ IamStack:
...
+  :::code{language=shell showLineNumbers=true showCopyAction=true}
+  cd cfn101-workshop/code/workspace/nested-stacks
+  :::
+1. Run the below command to copy the `iam.yaml` to S3 bucket. Replace the `bucket-name` with the bucket name from previous step.
+  :::code{language=shell showLineNumbers=false showCopyAction=true}
+  aws s3 cp iam.yaml s3://{bucket-name}
+  :::
+::::
+::::tab{id="local" label="Local development"}
+1. Navigate to the [S3 console](https://console.aws.amazon.com/s3/home?region=eu-west-1) and select your bucket.
```
Comment (cjhar) [inline content/intermediate/templates/nested-stacks/index.md:281]:
```suggestion
1. Navigate to the [S3 console](https://console.aws.amazon.com/s3/home) and select your bucket.
```

**Review by cjhar — DISMISSED** [review]
Some minor edits, fix one internal link and remove `eu-west-1` from console links

File: content/intermediate/templates/nested-stacks/index.md (hunk 2 above)
Comment (rezabekf) [reply content/intermediate/templates/nested-stacks/index.md:569]: Nice catch, yeah lets do that...

**Review by rezabekf — APPROVED** [review]
Great work all, this was the long one!

**Pull request merged**